streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -246,9 +246,7 @@ class StreamingContext private[streaming] (
     checkpointDir != null
   }
 
-  private[streaming] def initialCheckpoint: Checkpoint = {
-    if (isCheckpointPresent) _cp else null
-  }
+  private[streaming] def initialCheckpoint: Checkpoint = _cp
```

Comment:
Sorry @srowen to keep you waiting. As to the line, look at `isCheckpointPresent` and see that it does what the `if...else` block does, i.e. checks whether `_cp` is `null` and if it is, it returns `false` so `null`. It's a false check here and hence an easy fix (and that's why I proposed it with the other changes).

I can revert it and report another change if you want. Let me know.